app/controllers/assignments_controller.rb

```diff
@@ -61,7 +61,10 @@ def new_assignment_params
   end
 
   def set_assignment
+    @assignment = @organization.assignments.find_by!(id: params[:id])
+  rescue ActiveRecord::RecordNotFound
     @assignment = @organization.assignments.find_by!(slug: params[:id])
+    redirect_to action: params[:action], id: @assignment.id, organization_id: params[:organization_id]
```

Comment:
be default `redirect_to` is a `302` Found status.

Let's make this a `301` Permanently moved because this is a permanent change.

You can do this by.

```ruby
redirect_to action: params[:action], id: @assignment.id, organization_id: params[:organization_id], status: 302
```